tools/protoc-gen-docs/htmlGenerator.go

```diff
@@ -0,0 +1,869 @@
+		g.print("<!-- Generated by protoc-gen-docs -->")
+		g.print("<head>")
+		g.print("<meta charset='utf-8'>")
+		g.print("<meta name='viewport' content='width=device-width, initial-scale=1, shrink-to-fit=no'>")
+
+		if pkg.title != "" {
+			g.print("<meta name='title' content='", pkg.title, "'>")
+			g.print("<meta name='og:title' content='", pkg.title, "'>")
+			g.print("<title>", pkg.title, "</title>")
+		}
+
+		if pkg.overview != "" {
+			g.print("<meta name='description' content='", pkg.overview, "'>")
+			g.print("<meta name='og:description' content='", pkg.overview, "'>")
+		}
+
+		g.print(htmlStyle)
+
+		g.print("</head>")
+		g.print("<body>")
+		if pkg.title != "" {
+			g.print("<h1>", pkg.title, "</h1>")
+		}
+	} else if g.mode == htmlFragment {
+		g.print("<!-- Generated by protoc-gen-docs -->")
+		if pkg.title != "" {
+			g.print("<h1>", pkg.title, "</h1>")
+		}
+	}
+
+	g.print()
+
+	count := 0
+	for _, loc := range pkg.loc {
+		if loc.LeadingComments != nil {
+			g.generateComment(loc, "")
+			count++
+		}
+	}
+
+	if count == 0 {
+		g.warn(nil, "no comment for package %s", pkg.name)
+	}
+}
+
+func (g *htmlGenerator) generateFileFooter() {
+	if g.mode == htmlPage {
+		g.print("</body>")
+		g.print("</html>")
+	}
+}
+
+func (g *htmlGenerator) generateSectionHeading(name string) {
+	if g.grouping {
+		g.print("<h3 id='", name, "'>", name, "</h3>")
+	} else {
+		g.print("<h2 id='", name, "'>", name, "</h2>")
+	}
+	g.print("<section>")
+	g.print()
+}
+
+func (g *htmlGenerator) generateSectionTrailing() {
+	g.print("</section>")
+	g.print()
+}
+
+func (g *htmlGenerator) generateMessage(message *messageDescriptor) {
+	g.generateSectionHeading(g.relativeTypeName(message))
+
+	g.generateComment(message.loc, message.GetName())
+	g.print()
+
+	if len(message.fields) > 0 {
+		g.print("<table>")
+		g.print("<tr>")
+		g.print("<th>Field</th>")
+		g.print("<th>Type</th>")
+		g.print("<th>Description</th>")
+		g.print("</tr>")
+
+		var oneof int32 = -1
+		for _, field := range message.fields {
+			fieldName := camelCase(*field.Name)
+			fieldTypeName := g.fieldTypeName(field)
+
+			if field.OneofIndex != nil {
+				class := ""
+				if field.Options != nil && field.Options.GetDeprecated() {
+					class = "deprecated "
+				}
+
+				if *field.OneofIndex != oneof {
+					class = class + "oneof oneof-start"
+					oneof = *field.OneofIndex
+				} else {
+					class = class + "oneof"
+				}
+				g.print("<tr class='", class, "'>")
+			} else {
+				if field.Options != nil && field.Options.GetDeprecated() {
+					g.print("<tr class='deprecated' title='Deprecated'>")
+				} else {
+					g.print("<tr>")
+				}
+			}
+
+			g.print("<td><code>", fieldName, "</code></td>")
+			g.print("<td><code>", g.linkify(field.typ, fieldTypeName), "</code></td>")
+			g.print("<td>")
+
+			g.generateComment(field.loc, field.GetName())
+
+			g.print("</td>")
+			g.print("</tr>")
+		}
+		g.print("</table>")
+
+		/*
```

Comment:
dead code

Reply:
Not so much dead code as it is embryonic :-)

I want to add explicit support for a YAML-friendly description of each proto and that's what this is starting to be. I need to figure out what I actually want it to look like though before writing the code for it. I can finish this later.